### PR TITLE
[HardwareSecurityModules] Generating SDK

### PR DESCRIPTION
**Pull request by emmeliaAra**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/OutboundEnvironmentEndpointCollection.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Core;
+
+namespace Azure.ResourceManager.HardwareSecurityModules.Models
+{
+    /// <summary> Collection of OutboundEnvironmentEndpoint. </summary>
+    internal partial class OutboundEnvironmentEndpointCollection
```
Comment (archerzz) [inline sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/OutboundEnvironmentEndpointCollection.cs:16]: Avoid a module name ending with `Collection` to differentiate it from resource collection classes. Try to rename it to `OutboundEnvironmentEndpointGroup` or `OutboundEnvironmentEndpointList`

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/CloudHsmClusterData.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        }
+
+        /// <summary> The Cloud HSM Cluster's provisioningState. </summary>
+        public ProvisioningState? ProvisioningState { get; set; }
+        /// <summary> The Cloud HSM Cluster's auto-generated Domain Name Label Scope. </summary>
+        public string AutoGeneratedDomainNameLabelScope { get; set; }
+        /// <summary> Security domain properties information for Cloud HSM cluster. </summary>
+        public CloudHsmClusterSecurityDomainProperties SecurityDomain { get; set; }
+        /// <summary> An array of Cloud HSM Cluster's HSMs. </summary>
+        public IList<CloudHsmProperties> Hsms { get; }
```
Comment (archerzz) [inline sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/CloudHsmClusterData.cs:61]: This property name is a little bit weird. Can we rename it to `HsmProperties` or something similar?

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/DedicatedHsmData.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            Sku = sku;
+            Zones = zones;
+        }
+
+        /// <summary> Specifies the network interfaces of the dedicated hsm. </summary>
+        public NetworkProfile NetworkProfile { get; set; }
+        /// <summary> Specifies the management network interfaces of the dedicated hsm. </summary>
+        public NetworkProfile ManagementNetworkProfile { get; set; }
+        /// <summary> This field will be used when RP does not support Availability zones. </summary>
+        public string StampId { get; set; }
```
Comment (archerzz) [inline sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/DedicatedHsmData.cs:58]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/HardwareSecurityModulesPrivateEndpointConnectionData.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            get => PrivateEndpoint is null ? default : PrivateEndpoint.Id;
+        }
+
+        /// <summary> A collection of information about the state of the connection between service consumer and provider. </summary>
+        public HardwareSecurityModulesPrivateLinkServiceConnectionState ConnectionState { get; set; }
+        /// <summary> The provisioning state of the private endpoint connection resource. </summary>
+        public HardwareSecurityModulesPrivateEndpointConnectionProvisioningState? ProvisioningState { get; }
+        /// <summary> The group ids for the private endpoint resource. </summary>
+        public IReadOnlyList<string> GroupIds { get; }
```
Comment (archerzz) [inline sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/HardwareSecurityModulesPrivateEndpointConnectionData.cs:63]: What's the format of the element of this list?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/HardwareSecurityModulesPrivateLinkResource.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        /// <param name="requiredZoneNames"> The private link resource private link DNS zone name. </param>
+        internal HardwareSecurityModulesPrivateLinkResource(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, string groupId, IReadOnlyList<string> requiredMembers, IList<string> requiredZoneNames) : base(id, name, resourceType, systemData)
+        {
+            GroupId = groupId;
+            RequiredMembers = requiredMembers;
+            RequiredZoneNames = requiredZoneNames;
+        }
+
+        /// <summary> The private link resource group id. </summary>
+        public string GroupId { get; }
```
Comment (archerzz) [inline sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/HardwareSecurityModulesPrivateLinkResource.cs:40]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/NetworkInterface.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        /// <param name="id"> The ARM resource id in the form of /subscriptions/{SubscriptionId}/resourceGroups/{ResourceGroupName}/... </param>
+        /// <param name="privateIPAddress"> Private Ip address of the interface. </param>
+        internal NetworkInterface(string id, string privateIPAddress)
+        {
+            Id = id;
+            PrivateIPAddress = privateIPAddress;
+        }
+
+        /// <summary> The ARM resource id in the form of /subscriptions/{SubscriptionId}/resourceGroups/{ResourceGroupName}/... </summary>
+        public string Id { get; }
```
Comment (archerzz) [inline sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/NetworkInterface.cs:28]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

**Review by archerzz — COMMENTED** [review]
Generally, looks good. Need to confirm data types of some Ids.

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/DedicatedHsmData.cs (hunk 3 above)
Comment (emmeliaAra) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/DedicatedHsmData.cs:58]: StampId is just a string like "1" that our RP will use when the region does not support availability zones

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/NetworkInterface.cs (hunk 6 above)
Comment (emmeliaAra) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/NetworkInterface.cs:28]: NetworkInterface Id is a string like "/subscriptions/{subscriptionId}/resourceGroups/{resourcegroupID}/providers/Microsoft.Network/networkInterfaces/{nicName}"

**Comment by emmeliaAra** [comment]
@archerzz not sure why tests suddenly started to fail but none of these tests are test cases under HardwareSecurityModule SDK. 
image

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/DedicatedHsmData.cs (hunk 3 above)
Comment (archerzz) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/DedicatedHsmData.cs:58]: Got it. Thanks.

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/NetworkInterface.cs (hunk 6 above)
Comment (archerzz) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/NetworkInterface.cs:28]: > NetworkInterface Id is a string like "/subscriptions/{subscriptionId}/resourceGroups/{resourcegroupID}/providers/Microsoft.Network/networkInterfaces/{nicName}"

@emmeliaAra In that case, it should be a `ResourceIdentifier`. Need to change the format. See the guide link above.

Comment (emmeliaAra) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/NetworkInterface.cs]: @archerzz I am not able to open the guide links but I was able to make the change based on examples from other projects. Please let me know if there is any issue with the changes.

**Comment by emmeliaAra** [comment]
Hi @archerzz - There is a breaking-change label added to the PR, do we need to do anything to resolve this? 

Also if this PR is automatically merged once review is completed can we add the Do Not Merge label? We will like to keep this PR open because of some last minute changes in Swagger and we will need to revisit again once those are completed. 

-Thank you.

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/NetworkInterface.cs (hunk 6 above)
Comment (archerzz) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/NetworkInterface.cs]: LGTM Thanks!

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/README.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
+# Microsoft Azure Hardware Security Modules management client library for .NET
+
+**[Describe the service briefly first.]**
```
Comment (ArthurMa1978) [inline sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/README.md:3]: Please provide the description of your service here

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# Release History
+
+## 1.0.0-beta.1 (Unreleased)
```
Comment (ArthurMa1978) [inline sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/CHANGELOG.md:3]: Please provide the release date by replacing the `Unreleased`, format is `yyyy-mm-dd`

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/README.md (hunk 7 above)
Comment (emmeliaAra) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/README.md]: Description added

File: sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/CHANGELOG.md (hunk 8 above)
Comment (emmeliaAra) [reply sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/CHANGELOG.md]: Updated the release date, can we make this RP as Do not merge, please ?

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**